ktor-http/ktor-http-cio/jvm/src/io/ktor/http/cio/websocket/DefaultWebSocketSessionImpl.kt

```diff
@@ -92,11 +92,13 @@ class DefaultWebSocketSessionImpl(
         IncomingProcessorCoroutineName + Dispatchers.Unconfined
     ) {
         var last: BytePacketBuilder? = null
+        var closeFrameSent = false
         try {
             raw.incoming.consumeEach { frame ->
```

Comment:
`outgoing` or `filtered` cancellation should cancel `raw.incoming`

Reply:
consider adding a test to verify the correctness of `cancel()` instead of `terminate()`